R/module_filter_manager.R

```diff
@@ -137,22 +137,29 @@ filter_manager_srv <- function(id, filtered_data_list, filter) {
       }
 
     # Create mapping fo filters to modules in matrix form (presented as data.frame).
+    # Modules get NAs for filteres that cannot be set for them.
     mapping_matrix <- reactive({
-      module_states <- lapply(filtered_data_list, function(x) x$get_filter_state())
-      mapping_ragged <- lapply(module_states, function(x) vapply(x, `[[`, character(1L), "id"))
-      all_names <- vapply(slices_global(), `[[`, character(1L), "id")
-      mapping_smooth <- lapply(mapping_ragged, is.element, el = all_names)
-      as.data.frame(mapping_smooth, row.names = all_names, check.names = FALSE)
+      state_ids_global <- vapply(slices_global(), `[[`, character(1L), "id")
+      mapping_smooth <- lapply(filtered_data_list, function(x) {
+        state_ids_local <- vapply(x$get_filter_state(), `[[`, character(1L), "id")
+        state_ids_allowed <- vapply(x$get_available_teal_slices()(), `[[`, character(1L), "id")
```

Comment:
I'm just curious if this should be isolated (red one one a graph)? `FilteredData$available_teal_slices` is triggered by `slices_global`, so it seams to be redundant. I'm also worried that this reactivity graph might get out of control if we introduce new reactives. I think that making `eventReactive` might be a better solution to limit people in the future to trigger this by other reactives. Same here https://github.com/insightsengineering/teal.slice/pull/408#discussion_r1279336853

What do you think?


```suggestion
        state_ids_allowed <- vapply(isolate(x$get_available_teal_slices()()), `[[`, character(1L), "id")
```

![local-filter-panel-Page-13](https://github.com/insightsengineering/teal/assets/6959016/a1392e82-20a3-455d-9cb9-4d7a31b16a0c)

Reply:
Sorry, above is not true. Mapping matrix is displayed only when the filter manager is seen, so it doesn\t really matter.